openedx/openedx-platform
# feat: add language auto-tagging

## Description
This PR adds an auto-tagging feature (language) for Courses and XBlocks.

Every time a Course/XBlock is updated, if there is no Language ObjectTag associated with this Course/XBlock, a new one is created using the Course Language as a reference.

## Supporting information:
* Closes https://github.com/openedx/modular-learning/issues/79

## More Info
Depends on:
* https://github.com/openedx/openedx-learning/pull/71
* https://github.com/openedx/edx-platform/pull/32871

## Testing instructions

* Check the unit tests in `test_taks.py`.
* Run the unit tests

Manual check can be done in the CMS App:
1. Start `lms` and `studio` using `make dev.up.lms+studio+frontend-app-learning`
2. Enabled the `CONTENT_TAGGING_AUTO` CourseWaffleFlag
3. Open studio in `http://localhost:18010/` and login with `staff@example.com` / `edx`
4. Create a course
5. Create a course
6. Open cms python shell (`./in cms python manage.py cms shell`) and and check the created ObjectTags:
```python
from openedx_tagging.core.tagging.models import ObjectTag
object_tags = ObjectTag.objects.all()
object_tags
```

---
Private-ref: [FAL-3460](https://tasks.opencraft.com/browse/FAL-3460)
